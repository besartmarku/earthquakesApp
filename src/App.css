.App {
  text-align: center;
  background-color: #f9f8fd;
  padding: 80px;
  height: 100vh;
}
