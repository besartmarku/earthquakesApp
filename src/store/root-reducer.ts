import earthquakes from "./earthquakes/earthquakes.reducer";

const reducers = {
  earthquakes: earthquakes,
};

export default reducers;
