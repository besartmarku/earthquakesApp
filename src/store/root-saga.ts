import { all } from "redux-saga/effects";
import earthquakesSaga from "./earthquakes/earthquakes.saga";

export default function* rootSaga(): Generator {
  yield all([earthquakesSaga()]);
}
