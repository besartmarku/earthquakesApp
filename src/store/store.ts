import { applyMiddleware, combineReducers, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import logger from "redux-logger"; //dev purpose
import createSagaMiddleware from "redux-saga";
import reducers from "./root-reducer";
import rootSaga from "./root-saga";

const rootReducer = combineReducers({
  ...reducers,
});

export type RootState = ReturnType<typeof rootReducer>;

const sagaMiddleware = createSagaMiddleware();

const middleware = [sagaMiddleware];

const ENVIRONMENT = "development";

if (ENVIRONMENT === "development") {
  middleware.push(logger);
}

const enhancer = composeWithDevTools(applyMiddleware(...middleware));

const store = createStore(rootReducer, enhancer);

sagaMiddleware.run(rootSaga);

export { store };
