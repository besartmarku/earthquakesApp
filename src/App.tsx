import { useEffect } from "react";
import "./App.css";
import { useDispatch, useSelector } from "react-redux";
import {
  loadEarthquakes,
  filterEarthuqakes,
} from "./store/earthquakes/earthquakes.actions";
import { RootState } from "./store/store";

import { LoaderContainer, Loader } from "./components/shared/styles";
import SearchInput from "./components/shared/SearchInput";
import ListItem from "./components/shared/ListItem";

function App() {
  const dispatch = useDispatch();

  const {
    earthquakes: { loading, filteredEarthquakesList },
  } = useSelector((state: RootState) => state);

  useEffect(() => {
    dispatch(loadEarthquakes());
  }, []);

  const filterItems = (type: string, value?: string) => {
    dispatch(filterEarthuqakes({ type, value }));
  };

  return (
    <div className="App">
      <SearchInput
        placeholder="Enter Magnitude Value"
        buttonTitle="Search by magnitude"
        type="mag"
        onSearch={filterItems}
      />
      <br />

      <SearchInput
        placeholder="Enter magType Value"
        buttonTitle="Search by magType"
        type="magType"
        onSearch={filterItems}
      />
      <br />

      {loading ? (
        <LoaderContainer>
          <Loader />
        </LoaderContainer>
      ) : filteredEarthquakesList.length ? (
        filteredEarthquakesList.map((earthquake) => (
          <ListItem
            key={earthquake.id}
            id={earthquake.id}
            place={earthquake.properties.place}
            mag={earthquake.properties.mag}
            magType={earthquake.properties.magType}
          />
        ))
      ) : (
        <h2> No items founds </h2>
      )}
    </div>
  );
}

export default App;
