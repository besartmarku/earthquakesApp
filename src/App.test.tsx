import { render, screen } from "@testing-library/react";
import App from "./App";
import { store } from "./store/store";
import { Provider } from "react-redux";

import { RenderResult } from "@testing-library/react";

let wrapper: RenderResult;

describe("Homepage", () => {
  const mount = () => (
    <Provider store={store}>
      <App />
    </Provider>
  );

  describe("Homepage", () => {
    beforeEach(() => {
      wrapper = mount();
    });

    it("renders correctly", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
