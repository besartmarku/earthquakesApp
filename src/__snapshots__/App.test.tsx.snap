// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Homepage renders correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <App />
</Provider>
`;
